docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,240 @@ Note that:
-  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
\ No newline at end of file
+  then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+* Notifications for child-devices and their external devices are handled differently, as configuration files are fetched/pushed from/to external devices via HTTP filetransfer. Thereby notifications are part of the filetransfer contract (details see section [Fetching/Pushing configuration file](#fetchingpushing-configuration-file-fromto-the-external-device) below).
+
+## Configuration files for child devices
+
+To manage configuration files for child-devices the `c8y_configuration_plugin` aspects as below are relevant:
+
+* Associating with cloud's child-device twin, i.E. managing supported configuration list of child devices 
+  * an external child device declares it's supported config-types to thin-edge via MQTT
+  * thin-edge maintaines the supported config-types of each child-device a separate config-file
+  * the plugin is the one that associates the reported config-type list from the external to it's cloud device twin
+  * these config-list files per child-device is also managed by Configuration Management feature from C8Y
+* Consuming/Providing files from/to thin-edge via network
+  * any configuration file exchange to/from the external child device is handled by an HTTP based filetransfer service provided by thin-edge
+
+## Managing Supported Configuration List of child devices 
+
+The external device sends an MQTT message to `tedge/meta/plugin/configuration/<childid>` to announce it's configuration management capability to thin-edge. That MQTT message contains all configurations the external device provides. Thereby each configuration appears with a `type` and with an optional field `path`.
+
+The MQTT message is as below:
+
+```json
+{
+   "configurations": [
+     {
+       "type": "<config type 1>",
+       "path": "</path/to/file/file 1>"
+     },
+     {
+       "type": "<config type 2>",
+       "path": "</path/to/file/file 2>"
+     },
+     ...
+   ]
+}
+```
+
+Example:
+```json
+{
+   "configurations": [
+     {
+       "type": "foo.conf",
+       "path": "/etc/child1/foo.conf"
+     },
+     {
+       "type": "bar.conf",
+       "path": "/etc/child1/bar.conf"
+     }
+   ]
+}
+```
+
+If path is specified, the `c8y_configuration_plugin` consumes/provides the configuration-file from/to the given local filesystem path. If the field `path` is not given, the `c8y_configuration_plugin` make use of the HTTP filetransfer feature of the `tedge_agent` to consume/provide the configuration-file (see [section below](#details-to-aspect-2-filetransfer-fromto-external-device) for more details about HTTP filetransfer). The first case is intended for local processes (running on the thin-edge device) that represent a child-device, and the latter case is intended for external devices.
+
```

Comment:
From the above paragraph it's not clear that you expect **only** the "logical child devices" running on the thin-edge devices itself, to include the `path` field in its payload and **not** an "external child device".

Reply:
What do you mean with "logical child devices"? A local process (running on the thin-edge device) that represent a child-device?

Reply:
Yes. Just a shorter version of " local process running on the thin-edge device that represent a child-device". It would be nice if we have a clear terminology for such child device processes in the main child device document that we can use throughout our documentation.

Reply:
With removing the field path that entire paragraph was removed.

Reply:
There is now a terminology section, that also explains the term "logical child devices", see https://github.com/thin-edge/thin-edge.io/blob/65b88e1bd617b37478820d74a939379cb4972df5/docs/src/references/child-devices.md#terminology